frontend/js/index.js

```
@@ -1,12 +1,70 @@
-{
+document.addEventListener('DOMContentLoaded', () => {
+	const USERNAME_REC = 'username';
+
 	const container = document.querySelector('.messages');
+	const userNameContainer = document.querySelector('.dialog--username');
+	const dialogElement = document.querySelector('dialog--message');
+	const logoutButton = document.querySelector('.header__action--btn');
+	const editor = new EditorJS({
+		holder: 'editorjs',
+		tools: {
+			header: Header,
+			inlineCode: InlineCode,
+			paragraph: {
+				class: Paragraph,
+				inlineToolbar: true,
+			},
+		},
+		placeholder: 'Type your message here...',
+	});
+	const state = {
+		polling: false,
+		isRequestInProgress: false,
+		username: null,
+	};
+
+	async function poll() {
+		while (state.polling) {
+			await getMessages();
+			await new Promise((resolve) => setTimeout(resolve, 3000));
+		}
+	}
+
+	function startPolling() {
+		if (!state.polling) {
+			state.polling = true;
+			poll();
+		}
+	}
+
+	function stopPolling() {
+		state.polling = false;
+	}
+
+	function validateMessage(message) {
+		if (message.length < 2) {
```

I like fail fas. But I think fail fast is more appropriate when function returns something else apart from booleans, like, messages for instance. If it's only booleans you can simplify it:

```js
function isMessageValid(text) {
  return (
    message.length >= 2 &&
    message.length <= 400
  );
}
```

Or:
```js
function getValidationMessage(text) {
  if (text.length < 2) {
    return "Text is too short";
  }
  
  if (text.length > 400) {
    return "Text is too long";
  }
  
  return null;
}
```

In this case you'll be using it like this:
```js
const errorMessage = getValidationMessage(msg);

if (errorMessage !== null) {
  //...
  alert("");
  setBuildingOnFire();
  call911();
}

smile();
```